onnxruntime/python/onnxruntime_pybind_state.cc

```diff
@@ -324,6 +330,42 @@ const std::vector<std::string>& GetAvailableProviders() {
   return available_providers;
 }
 
+#ifdef USE_CUDA
+bool is_cuda_device_id_valid(int id) {
+    
+    int num_devices = 0;
+    CUDA_CALL_THROW(cudaGetDeviceCount(&num_devices));
+
+    if (0 == num_devices)
+    {
+        printf("your system does not have a CUDA capable device.\n");
+        return false;
+    }
+
+    if (id < 0 || id >= num_devices)
+    {
+        printf("cuda_device=%d is invalid, must choose device ID between 0 and %d\n", id, num_devices - 1);
+        return false;
+    }
+
+    return true;
+}
+
+void update_cuda_provider_options(onnxruntime::CudaProviderOptions& options, std::map<std::string, std::string> options_map) {
+
+    auto it = options_map.find("device_id");
```

Comment:
this doesn't seem to handle updating the other cuda options
such as cuda_mem_limit and arena_extend_strategy ?

Reply:
I've added codes to support updating these two cuda options.